File: 0735-asteroid-collision/0735-asteroid-collision.js
/**
 * @param {number[]} asteroids
 * @return {number[]}
 */
var asteroidCollision = function(asteroids) {
    let ans = [], flag = 1;
    for(let i=0;i<asteroids.length;i++) {
        if(ans.length && ans.slice(-1)>0 && asteroids[i]<0) {
            flag = 1;
            while(ans.length && ans.slice(-1)>0 && asteroids[i]<0) {
            if(Math.abs(ans.slice(-1))<Math.abs(asteroids[i])) {
                ans.pop();
            } else if(Math.abs(ans.slice(-1))===Math.abs(asteroids[i])) {
                flag=0;
                ans.pop();
                break;
            } else {
                flag=0;
                break;
               }
        }
            if(flag) ans.push(asteroids[i]);
        } else {
            ans.push(asteroids[i]);
        }
    }
    return ans;
};

File: 0735-asteroid-collision/0735-asteroid-collision.cpp
class Solution {
public:
    vector<int> asteroidCollision(vector<int>& nums) {
        vector<int> ans;
        bool pushFlag = 1;
        int n = nums.size(), num1, num2;
        for(int i=0;i<n;i++) {
           if(ans.size() && ans.back()>0 && nums[i]<0) {
               pushFlag = 1;
               while(ans.size() && ans.back()>0 && nums[i]<0) {
                   num1 = abs(ans.back()), num2 = abs(nums[i]);
                   if(num1 < num2) {ans.pop_back();}
                   else if(num1 == num2) {
                       ans.pop_back();
                       pushFlag = 0;
                       break;
                   } else { pushFlag = 0; break;}
               }
               if(pushFlag) { ans.push_back(nums[i]); }
           } else {
               ans.push_back(nums[i]);
           }
        }
        return ans;
    }
};